pyrit/scenarios/scenario.py

```diff
@@ -176,28 +193,231 @@ def get_default_strategy(cls) -> ScenarioStrategy:
 
     async def initialize_async(self) -> None:
         """
-        Initialize the scenario by populating self._atomic_attacks
+        Initialize the scenario by populating self._atomic_attacks and creating the ScenarioResult.
 
         This method allows scenarios to be initialized with atomic attacks after construction,
         which is useful when atomic attacks require async operations to be built.
 
+        If a scenario_result_id was provided in __init__, this method will check if it exists
+        in memory and validate that the stored scenario matches the current configuration.
+        If it matches, the scenario will resume from prior progress. If it doesn't match or
+        doesn't exist, a new scenario result will be created.
+
         Args:
             atomic_attacks: List of AtomicAttack instances to execute in this scenario.
 
         Returns:
             Scenario: Self for method chaining.
 
         Example:
+            >>> # New scenario
             >>> scenario = MyScenario(
             ...     objective_target=target,
             ...     attack_strategies=["base64", "leetspeak"]
             ... )
-            >>> atomic_attacks = await scenario.build_atomic_attacks_async()
             >>> await scenario.initialize_async()
             >>> results = await scenario.run_async()
+            >>>
+            >>> # Resume existing scenario
+            >>> scenario_id = results.id
+            >>> resumed_scenario = MyScenario(
+            ...     objective_target=target,
+            ...     attack_strategies=["base64", "leetspeak"],
+            ...     scenario_result_id=str(scenario_id)
+            ... )
+            >>> await resumed_scenario.initialize_async()
+            >>> results = await resumed_scenario.run_async()  # Resumes from progress
         """
         self._atomic_attacks = await self._get_atomic_attacks_async()
 
+        # Store original objectives for each atomic attack (before any mutations during execution)
+        self._original_objectives_map = {
+            atomic_attack.atomic_attack_name: list(atomic_attack._objectives) for atomic_attack in self._atomic_attacks
```

Comment:
```suggestion
            atomic_attack.atomic_attack_name: tuple(atomic_attack._objectives) for atomic_attack in self._atomic_attacks
```
tuple is immutable by default, helps prevent original values from changing, but should we use a dataclass to freeze this just in case?